lib/route-recognizer.js

```diff
@@ -89,13 +89,13 @@ EpsilonSegment.prototype = {
   generate: function() { return ""; }
 };
 
-function parse(route, names, specificity) {
+function parse(allSegments, route, names, specificity) {
   // normalize route as not starting with a "/". Recognition will
   // also normalize.
   if (route.charAt(0) === "/") { route = route.substr(1); }
 
   var segments = route.split("/");
-  var results = new Array(segments.length);
+  //var results = new Array(segments.length);
```

Comment:
This was nearly half the improvement, the regex laziness another small piece, the number to string to number issue needs a more realistic benchmark to see. The benchmark here actually is pretty bad. It would be better to take an app with a lot of routes and save the calls to add and redo this.  I expected the specificity to be more.